### PR TITLE
Improvement: Abstract cors configuration away from the user, and just allow them to turn it on or off.

### PR DESCRIPTION
**Pull request by bennovakovic**

### CODE REVIEW

File: src/PKCE.ts [hunk 1]
```diff
@@ -19,6 +21,20 @@ export default class PKCE {
     this.config = config;
   }
 
+  /**
+   * Allow the user to enable cross domain cors requests
+   * @param  enable turn the cross domain request options on.
+   * @return Promise<ICorsOptions>
```
Comment (bpedroza) [inline src/PKCE.ts:27]: This does not return a promise.

File: src/PKCE.ts [hunk 2]
```diff
@@ -71,6 +87,7 @@ export default class PKCE {
           Accept: 'application/json',
           'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
         },
+        ...this.corsRequestOptions
```
Comment (bpedroza) [inline src/PKCE.ts:90]: Please include this new code in the `refreshAccessToken` method.

File: tests/PKCE.test.ts [hunk 3]
```diff
@@ -206,3 +207,17 @@ describe('Test storage types', () => {
     expect(localStorage.getItem('pkce_code_verifier')).not.toEqual(null);
   });
 });
+
+describe('Test additional cors options', () => {
+  it('Should have set the cors options correctly', async () => {
+    const instance = new PKCE(config);
+    const opts = instance.enableCorsCredentials(true)
+    expect(opts.credentials).toEqual('include')
+    expect(opts.mode).toEqual('cors')
```
Comment (bpedroza) [inline tests/PKCE.test.ts:216]: These tests are not testing that the code is working when making requests. Please update to run tests against mocked fetch calls.

**Review by bpedroza — CHANGES_REQUESTED** [review]

File: src/PKCE.ts (hunk 2 above)
Comment (bennovakovic) [reply src/PKCE.ts:90]: As previously discussed [here](https://github.com/bpedroza/js-pkce/pull/36#issuecomment-1709358984), this is not required.

File: tests/PKCE.test.ts (hunk 3 above)
Comment (bennovakovic) [reply tests/PKCE.test.ts]: Pushed a change to enable this. I placed it in the existing group and extended the mockRequest in that group to take an additional optional parameter. This was just to reduce the amount of mocked code.

**Review by bpedroza — APPROVED** [review]

**Pull request merged**